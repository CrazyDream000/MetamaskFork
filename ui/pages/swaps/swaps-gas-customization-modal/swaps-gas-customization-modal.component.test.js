import React from 'react';

import { renderWithProvider } from '../../../../test/jest';
import GasCustomizationModalComponent from './swaps-gas-customization-modal.component';

const createProps = (customProps = {}) => {
  return {
    insufficientBalance: false,
    onSubmit: jest.fn(),
    cancelAndClose: jest.fn(),
    minimumGasLimit: 100,
    gasPriceButtonGroupProps: {
      buttonDataLoading: false,
      className: 'gas-price-button-group',
      gasButtonInfo: [
        {
          feeInPrimaryCurrency: '$0.52',
          feeInSecondaryCurrency: '0.0048 ETH',
          timeEstimate: '~ 1 min 0 sec',
          priceInHexWei: '0xa1b2c3f',
          gasEstimateType: 'FAST',
        },
        {
          feeInPrimaryCurrency: '$0.39',
          feeInSecondaryCurrency: '0.004 ETH',
          timeEstimate: '~ 1 min 30 sec',
          priceInHexWei: '0xa1b2c39',
          gasEstimateType: 'FAST',
        },
        {
          feeInPrimaryCurrency: '$0.30',
          feeInSecondaryCurrency: '0.00354 ETH',
          timeEstimate: '~ 2 min 1 sec',
          priceInHexWei: '0xa1b2c30',
          gasEstimateType: 'FAST',
        },
      ],
      handleGasPriceSelection: 'mockSelectionFunction',
      noButtonActiveByDefault: true,
      showCheck: true,
      newTotalFiat: 'mockNewTotalFiat',
      newTotalEth: 'mockNewTotalEth',
    },
    infoRowProps: {
      originalTotalFiat: 'mockOriginalTotalFiat',
      originalTotalEth: 'mockOriginalTotalEth',
      newTotalFiat: 'mockNewTotalFiat',
      newTotalEth: 'mockNewTotalEth',
      sendAmount: 'mockSendAmount',
      transactionFee: 'mockTransactionFee',
      extraInfoRow: { label: 'mockLabel', value: 'mockValue' },
    },
    ...customProps,
  };
};

describe('GasCustomizationModalComponent', () => {
  it('renders the component with initial props', () => {
    const props = createProps();
    const { getByText, getByTestId } = renderWithProvider(
      <GasCustomizationModalComponent {...props} />,
    );
    expect(getByTestId('page-container__header')).toMatchSnapshot();
    expect(getByText('$0.52')).toBeInTheDocument();
    expect(getByText('0.0048 ETH')).toBeInTheDocument();
    expect(getByTestId('button-group__button0')).toMatchSnapshot();
    expect(getByText('~ 1 min 30 sec')).toBeInTheDocument();
    expect(getByText('$0.39')).toBeInTheDocument();
    expect(getByText('0.004 ETH')).toBeInTheDocument();
    expect(
      getByTestId('gas-modal-content__info-row__send-info'),
    ).toMatchSnapshot();
    expect(
      getByTestId('gas-modal-content__info-row__transaction-info'),
    ).toMatchSnapshot();
    expect(
      getByTestId('gas-modal-content__info-row__total-info'),
    ).toMatchSnapshot();
    expect(getByText('Save')).toBeInTheDocument();
  });
});
