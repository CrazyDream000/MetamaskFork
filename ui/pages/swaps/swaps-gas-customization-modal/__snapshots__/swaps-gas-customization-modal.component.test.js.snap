// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GasCustomizationModalComponent renders the component with initial props 1`] = `
<div
  class="page-container__header page-container__header--no-padding-bottom"
  data-testid="page-container__header"
>
  <div
    class="page-container__title"
  >
    Customize Gas
  </div>
  <div
    class="page-container__subtitle"
  >
    Increasing fee may decrease processing times, but it is not guaranteed.
  </div>
  <a
    class="button btn-link page-container__header-close-text"
    role="button"
    tabindex="0"
  >
    Close
  </a>
  <ul
    class="page-container__tabs"
  >
    <li
      class="tab page-container__tab tab--active"
    >
      <button>
        Basic
      </button>
    </li>
    <li
      class="tab page-container__tab"
    >
      <button>
        Advanced
      </button>
    </li>
  </ul>
</div>
`;

exports[`GasCustomizationModalComponent renders the component with initial props 2`] = `
<button
  aria-checked="false"
  class="button-group__button"
  data-testid="button-group__button0"
>
  <div>
    <div
      class="gas-price-button-group__label"
    >
      Fast
    </div>
    <div
      class="gas-price-button-group__time-estimate"
    >
      ~ 1 min 0 sec
    </div>
    <div
      class="gas-price-button-group__primary-currency"
    >
      $0.52
    </div>
    <div
      class="gas-price-button-group__secondary-currency"
    >
      0.0048 ETH
    </div>
    <div
      class="button-check-wrapper"
    >
      <i
        class="fa fa-check fa-sm"
      />
    </div>
  </div>
</button>
`;

exports[`GasCustomizationModalComponent renders the component with initial props 3`] = `
<div
  class="gas-modal-content__info-row__send-info"
  data-testid="gas-modal-content__info-row__send-info"
>
  <span
    class="gas-modal-content__info-row__send-info__label"
  >
    Send Amount
  </span>
  <span
    class="gas-modal-content__info-row__send-info__value"
  >
    mockSendAmount
  </span>
</div>
`;

exports[`GasCustomizationModalComponent renders the component with initial props 4`] = `
<div
  class="gas-modal-content__info-row__transaction-info"
  data-testid="gas-modal-content__info-row__transaction-info"
>
  <span
    class="gas-modal-content__info-row__transaction-info__label"
  >
    Transaction Fee
  </span>
  <span
    class="gas-modal-content__info-row__transaction-info__value"
  >
    mockTransactionFee
  </span>
</div>
`;

exports[`GasCustomizationModalComponent renders the component with initial props 5`] = `
<div
  class="gas-modal-content__info-row__total-info"
  data-testid="gas-modal-content__info-row__total-info"
>
  <span
    class="gas-modal-content__info-row__total-info__label"
  >
    New Total
  </span>
  <span
    class="gas-modal-content__info-row__total-info__value"
  >
    mockNewTotalEth
  </span>
</div>
`;
